src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -70,7 +70,9 @@ SecurityAnalysisReport runSync(String workingVariantId, SecurityAnalysisParamete
         OpenLoadFlowParameters lfParametersExt = OpenLoadFlowParameters.get(securityAnalysisParameters.getLoadFlowParameters());
         // in some post-contingency computation, it does not remain elements to participate to slack distribution.
         // in that case, the remaining mismatch is put on the slack bus and no exception is thrown.
-        lfParametersExt.setThrowsExceptionInCaseOfSlackDistributionFailure(false);
+        if (lfParametersExt.getSlackDistributionFailureBehavior() == OpenLoadFlowParameters.SlackDistributionFailureBehavior.THROW) {
+            lfParametersExt.setSlackDistributionFailureBehavior(OpenLoadFlowParameters.SlackDistributionFailureBehavior.LEAVE_ON_SLACK_BUS); // or FAIL ?
```

Comment:
I figure out that we have an existing bug here. We can leave as it is in this PR and fix later if you want
 - we should not change the `LoadFlowParameters` in an AS run. `SecurityAnalysisParameters` and nested `LoadFlowParameters` are input data and in order of a security analysis run to be idempotent, parameters should not be modified.
 -  I think (to be discussed) that for the N loadflow we should keep input behavior (so it could throw an exception if requested) but for N-K and post action loadflow we should force to FAIL behavior and not run actions if slack bus distribution has failed in N-K LF.

Reply:
Agree. @geofjamg is there an existing way to tell AcLoadFlowEngine it is running an SA ? Cannot find it... is adding a boolean to AcLoadFlowContext the way to go ?

Thanks to your comment, I found out that in AcSecurityAnalysis `postContingencyComputationOk` and `PostContingencyComputationStatus` calculations are only looking at NR status, ignoring outerloop status. Hence it is not aligned with loadflow. Will work out a fix in this PR.

Reply:
`AcLoadFlowEngine` has not to know in which context it is running. From the SA code, you can configure it differently depending if it is ran in N, N-K or action.